all/Dockerfile

```diff
@@ -1,15 +1,30 @@
 FROM python:3.11.2-slim as build
+
 RUN set -ex \
   && apt-get -y update \
   && apt-get -y install --no-install-recommends git \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
 WORKDIR /opt/octodns
-COPY requirements.txt requirements.txt
+
+COPY requirements.txt all-requirements.txt
+
+ARG OCTODNS_FLAVOR=octodns
```

Comment:
I haven't seen this before. The arg will overwrite this? There's no `:=` needed like in bash?

Reply:
Yep, this will act as an ARG with a default value which is then exposed when building as an ENV, thus sets "octodns" as a default (in case of not set) yet allowing flavor to be overwritten at run time via `docker build --build-arg OCTODNS_FLAVOR=azure`.

https://docs.docker.com/engine/reference/builder/#default-values

Reply:
Great! Thanks for the doc reference too.